src/android/StatusReport.java

```diff
@@ -0,0 +1,80 @@
+package com.microsoft.cordova;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class StatusReport {
+
+    private static final String STATUS_KEY = "status";
+    private static final String LABEL_KEY = "label";
+    private static final String APP_VERSION_KEY = "appVersion";
+    private static final String DEPLOYMENT_KEY_KEY = "deploymentKey";
+    private static final String LAST_VERSION_LABEL_OR_APP_VERSION_KEY = "lastVersionLabelOrAppVersion";
+    private static final String LAST_VERSION_DEPLOYMENT_KEY_KEY = "lastVersionDeploymentKey";
+
+    ReportingStatus status;
+    String label;
+    String appVersion;
+    String deploymentKey;
+    String lastVersionLabelOrAppVersion;
+    String lastVersionDeploymentKey;
+
+    public StatusReport(int status, String label, String appVersion, String deploymentKey, String lastVersionLabelOrAppVersion, String lastVersionDeploymentKey) {
+        this(ReportingStatus.values()[status], label, appVersion, deploymentKey, lastVersionLabelOrAppVersion, lastVersionDeploymentKey);
+    }
+
+    public StatusReport(ReportingStatus status, String label, String appVersion, String deploymentKey) {
+        this(status, label, appVersion, deploymentKey, null, null);
+    }
+
+    public StatusReport(ReportingStatus status, String label, String appVersion, String deploymentKey, String lastVersionLabelOrAppVersion, String lastVersionDeploymentKey) {
+        this.status = status;
+        this.label = label;
+        this.appVersion = appVersion;
+        this.deploymentKey = deploymentKey;
+        this.lastVersionLabelOrAppVersion = lastVersionLabelOrAppVersion;
```

Comment:
could we check if `lastVersionLabelOrAppVersion` and `lastVersionDeploymentKey` are null and fill them with the values from `codePushPreferences` instead of just having a null references here?